src/main/java/gregtech/api/mui/synchandler/GhostCircuitSyncHandler.java

```diff
@@ -0,0 +1,86 @@
+package gregtech.api.mui.synchandler;
+
+import gregtech.api.capability.impl.GhostCircuitItemStackHandler;
+import gregtech.api.recipes.ingredients.IntCircuitIngredient;
+
+import net.minecraft.network.PacketBuffer;
+import net.minecraftforge.items.IItemHandler;
+
+import com.cleanroommc.modularui.utils.MouseData;
+import com.cleanroommc.modularui.value.sync.ItemSlotSH;
+import com.cleanroommc.modularui.widgets.slot.ModularSlot;
+
+import java.io.IOException;
+
+public class GhostCircuitSyncHandler extends ItemSlotSH {
+
+    public GhostCircuitSyncHandler(ModularSlot slot) {
+        super(slot);
+    }
+
+    @Override
+    protected void phantomClick(MouseData mouseData) {
+        if (mouseData.mouseButton == 0) {
+            // increment on left-click
+            setCircuitValue(getNextCircuitValue(1));
+        } else if (mouseData.mouseButton == 1 && mouseData.shift) {
+            // clear on shift-right-click
+            setCircuitValue(GhostCircuitItemStackHandler.NO_CONFIG);
+        } else if (mouseData.mouseButton == 1) {
+            // decrement on right-click
+            setCircuitValue(getNextCircuitValue(-1));
+        }
+    }
+
+    @Override
+    protected void phantomScroll(MouseData mouseData) {
+        setCircuitValue(getNextCircuitValue(mouseData.mouseButton));
+    }
+
+    private void setCircuitValue(int value) {
+        GhostCircuitItemStackHandler handler = getGhostCircuitHandler();
+        handler.setCircuitValue(value);
+        syncToClient(1, buf -> {
```

Comment:
Should there be a check to see if the value is going to be different before updating it and syncing the change?